pkg/sql/ttl/ttlbase/ttl_helpers.go

```diff
@@ -13,25 +13,90 @@ package ttlbase
 import (
 	"bytes"
 	"fmt"
+	"math"
 	"strconv"
 	"time"
 
+	"github.com/cockroachdb/cockroach/pkg/settings"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catenumpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/tabledesc"
 )
 
-// DefaultAOSTDuration is the default duration to use in the AS OF SYSTEM TIME
-// clause used in the SELECT query.
-const DefaultAOSTDuration = -time.Second * 30
+const (
+	// DefaultAOSTDuration is the default duration to use in the AS OF SYSTEM TIME
+	// clause used in the SELECT query.
+	DefaultAOSTDuration         = -time.Second * 30
+	DefaultSelectBatchSizeValue = 500
+)
+
+var (
+	defaultSelectBatchSize = settings.RegisterIntSetting(
+		settings.TenantWritable,
+		"sql.ttl.default_select_batch_size",
+		"default amount of rows to select in a single query during a TTL job",
+		DefaultSelectBatchSizeValue,
+		settings.PositiveInt,
+		settings.WithPublic,
+	)
+	defaultDeleteBatchSize = settings.RegisterIntSetting(
+		settings.TenantWritable,
+		"sql.ttl.default_delete_batch_size",
+		"default amount of rows to delete in a single query during a TTL job",
+		100,
+		settings.PositiveInt,
+		settings.WithPublic,
+	)
+	defaultDeleteRateLimit = settings.RegisterIntSetting(
+		settings.TenantWritable,
+		"sql.ttl.default_delete_rate_limit",
+		"default delete rate limit for all TTL jobs. Use 0 to signify no rate limit.",
```

Comment:
nit: can you specify the units on this or further describe how this number is interpreted?

edit: I see now this was just moved. Would be a nice to have to improve the documentation though :D

Reply:
I'll change it now anyways to make it more clear.